### DISCUSSION TITLE
Link Multiple OAuth Account (Same provider)

### DISCUSSION THREAD

**Discussion by DblK** [meteor/meteor#13006] [Feature Requests]
I saw that the oauth account are stored into `services` object and have a key by provider (`facebook`, `google`, etc..).
I was wondering why those entries are objects and not array?
If it were arrays, this will allow to link multiple oauth provider into the same user. For example, to link multiple google account to see all calendars, or multiple facebook account to make cross post easily.

Is there a catch of doing so (Beside the fact to update all oauth packages providers)?

**Comment by StorytellerCZ** [comment]
Beside the one that you have mentioned the accounts packages and OAuth connections are meant just for login and to be 1 to 1.
There is [link accounts](https://github.com/Meteor-Community-Packages/meteor-link-accounts) package that provides some additional functionality and also links to #1133 for one aspect of this discussion. The main point of the OAuth packages are for authentication, rather than integration. But I'm open to have the discussion about changing this if there is a good plan. One issue with integrations is that they are not necessarily linked to your account and sometimes one social media account can be linked to multiple accounts, so we would need a clear story here to communicate to people which approach we are taking.

**Comment by DblK** [comment]
I use also the package you mention and I faced the issue of 1 to 1.
However, if we do not treat them and authentication but, like you said, integration I think we can manage to have something that might work.

So first, we can think about adding an object `integrations` side by side with `services`.
This will allow to see the separation with the two and forbid connection with the integrations services.
For the structure inside this, I will think about something that should address all problems.
But yeah, this will force integrations to be linked to a user. If we do not want that, then the users should provide the collection in which we will add the `integrations` object on each document on the collection.

Then, about the account to multiple users, we can 1) forbid it 2) allow it.
1) We will need to tell user to make something like teams or similar to be able to share their integration with other users (We can make an index on it then)
2) Allowing it, will provide more freedom for the users
1+2) Both, then we will let the user tells during config if it should be unique or not and we need to handle this.

Or we can add a new collection `integrations`, with an array of `userIds` to know which user has access to which integration.

Finally, about implementation, we can build something similar to `link account` to reuse some part of existing package to use them as integration instead of authentification.

Before going deep to code level, I agree with you we should make it clear to how to use it with pros and cons.

**Reply by DblK** [reply]
@StorytellerCZ any thought about that? I am willing to work on it btw.
